napari/_qt/widgets/qt_viewer_dock_widget.py

```diff
@@ -152,6 +153,22 @@ def __init__(
         self.setTitleBarWidget(self.title)
         self.visibilityChanged.connect(self._on_visibility_changed)
 
+        self.dockLocationChanged.connect(self._update_default_dock_area)
+
+    def _update_default_dock_area(self, value):
+        if value == Qt.DockWidgetArea.NoDockWidgetArea:
+            return
+        settings = get_settings()
+        if value == Qt.DockWidgetArea.LeftDockWidgetArea:
+            settings.application.plugins_widget_positions[self.name] = 'left'
+        elif value == Qt.DockWidgetArea.RightDockWidgetArea:
+            settings.application.plugins_widget_positions[self.name] = 'right'
+        elif value == Qt.DockWidgetArea.TopDockWidgetArea:
+            settings.application.plugins_widget_positions[self.name] = 'top'
+        elif value == Qt.DockWidgetArea.BottomDockWidgetArea:
+            settings.application.plugins_widget_positions[self.name] = 'bottom'
```

Comment:
These if/elif cascades always make me want to find a different way. This might not be better, but putting the suggestion down for comment:

```
area2str = {
    getattr(Qt.DockWidgetArea, pos.title() + 'DockWidgetArea') : pos
    for pos in ['left', 'right', 'top', 'bottom']
}

...

settings.application.plugins_widget_positions[self.name] = area2str[value]
```

Reply:
Once we update to python 3.10 we may use switch statement. 
I may create a global dict to workaround this syntax.